graphql-kotlin-schema-generator/src/main/kotlin/com/expediagroup/graphql/execution/FunctionDataFetcher.kt

```diff
@@ -64,7 +65,12 @@ open class FunctionDataFetcher(
             val parameterValues = getParameterValues(fn, environment)
 
             if (fn.isSuspend) {
-                runSuspendingFunction(it, parameterValues)
+                val scope = try {
+                    environment.getContext<CoroutineScope>()!!
+                } catch (e: Exception) {
+                    GlobalScope
+                }
```

Comment:
Instead of throwing and catching exceptions on null can this just be the following? I haven't looked at the method signatures fully

```suggestion
                val scope = environment.getContext<CoroutineScope>() ?: GlobalScope
```

Reply:
`environment.getContext<T>` looks like it just blindly does a cast to `T`, so would throw a ClassCastException if the context didn't implement scope.  maybe could do something like `(environment.getContext<Any>() as? CoroutineScope) ?: GlobalScope`, but not sure if that's better or worse?

Reply:
I think that is better than using `!!` as we can see what the code would be failing to do that returns null